src/controllers/main/main.ts

```diff
@@ -773,6 +798,66 @@ export class MainController extends EventEmitter {
     clearTimeout(timeoutId)
   }
 
+  // show signAccountOp with previously added userRequests kind calls
+  #makeAccountOpFromStackedCalls(networkId: string, accountAddr: string) {
+    // no account op request if there isn't one
+    const callsUserReq = this.userRequests.find(
+      (req) =>
+        req.action.kind === 'calls' &&
+        req.meta.networkId === networkId &&
+        req.meta.accountAddr === accountAddr
+    )
+    if (!callsUserReq) return
+
+    const accountOpAction = makeSmartAccountOpAction({
+      account: this.accounts.accounts.find((acc) => acc.addr === accountAddr)!,
+      networkId,
+      nonce: this.accounts.accountStates[accountAddr][networkId].nonce,
+      userRequests: this.userRequests,
+      actionsQueue: this.actions.actionsQueue
+    })
+
+    this.actions.addOrUpdateAction(accountOpAction, 'first')
```

Comment:
1. BA -> add new txn on Odyssey
2. The same BA -> add new txn on Odyssey
3. Sign 7702 auth message.
4. The first txn is removed and replaced with a new AccOp that contains the both transactions.
5. But, the second transaction is still part of the Actions queue.

I would suggest in that case, if we sign an 7702 auth message, first to stack the existing txns into an acc op (as we did it), but the same time to remove all txns those are on the network we authorized.

![Screenshot from 2025-02-20 17-46-38](https://github.com/user-attachments/assets/fb433483-14bd-45fe-8100-c10e9ded7d77)
![Screenshot from 2025-02-20 17-46-23](https://github.com/user-attachments/assets/3317db4a-10bd-4e0f-8fdc-f8560acfab0a)

Reply:
This flow is fixed for the time being, albeit ugly. We're changing the flow with the authorization message completely so I think this part of the code will get deleted. If not, let's discuss it at a later stage